sdk/openai/Azure.AI.OpenAI/src/Custom/Files/AzureFileClient.cs

```diff
@@ -30,4 +31,48 @@ internal partial class AzureFileClient : FileClient
 
     protected AzureFileClient()
     { }
+
+    /// <inheritdoc />
+    public override ClientResult<OpenAIFileInfo> UploadFile(Stream file, string filename, FileUploadPurpose purpose, CancellationToken cancellationToken = default)
```

Comment:
It's very problematic for us to override convenience methods, as it creates severe breaking change potential with the base library -- this was the fundamental reason we had the emergency fix last week associated with the CancellationToken change -- the overrides in ChatClient were invalidated by the signature update and broke the code path.

Do you think these changes could be appropriate to the underlying OpenAI client even if the behavioral difference is Azure-specific? E.g. I don't think setting the content type should be problematic (might even be preferrable, required or not) and then we may be able to keep the overloads limited to protocol methods.

Reply:
This is the workaround `MultipartFormDataBinaryContent` not setting the Content-Type header to text/plain. The OpenAI service doesn't care about this. Azure OpenAI does and returns HTTP 400. This is intended to be a short term workaround until the underlying issue is addressed. I also don't want to gate and delay this PR any further than it already has on other changes.